MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.ScrollViewer.xaml

```diff
@@ -10,15 +12,96 @@
+
+            <Rectangle 
+                x:Name="PART_Corner" 
+                Grid.Column="1" 
+                Fill="{DynamicResource MaterialDesignSelection}" 
+                Grid.Row="1"/>
+
+            <ScrollContentPresenter 
+                x:Name="PART_ScrollContentPresenter" 
+                CanContentScroll="{TemplateBinding CanContentScroll}"
+                CanHorizontallyScroll="False" CanVerticallyScroll="False"
+                ContentTemplate="{TemplateBinding ContentTemplate}"
+                Content="{TemplateBinding Content}"
+                Grid.Column="0" 
+                Margin="{TemplateBinding Padding}"
+                Grid.Row="0"/>
+
+            <ScrollBar 
+                x:Name="PART_VerticalScrollBar" 
+                AutomationProperties.AutomationId="VerticalScrollBar"
+                Cursor="Arrow" 
+                Grid.Column="1"
+                Maximum="{TemplateBinding ScrollableHeight}"
+                Minimum="0" 
+                Grid.Row="0" 
+                Visibility="{TemplateBinding ComputedVerticalScrollBarVisibility}" 
+                Value="{Binding VerticalOffset, Mode=OneWay, RelativeSource={RelativeSource TemplatedParent}}" 
+                ViewportSize="{TemplateBinding ViewportHeight}" />
+
+            <ScrollBar 
+                x:Name="PART_HorizontalScrollBar"
+                AutomationProperties.AutomationId="HorizontalScrollBar" 
+                Cursor="Arrow" 
+                Grid.Column="0" 
+                Maximum="{TemplateBinding ScrollableWidth}" 
+                Minimum="0" 
+                Orientation="Horizontal" 
+                Grid.Row="1" 
+                Visibility="{TemplateBinding ComputedHorizontalScrollBarVisibility}"
+                Value="{Binding HorizontalOffset, Mode=OneWay, RelativeSource={RelativeSource TemplatedParent}}" 
+                ViewportSize="{TemplateBinding ViewportWidth}" />
         </Grid>
+        <ControlTemplate.Triggers>
+            <Trigger Property="wpf:ScrollViewerAssist.CornerRectangleVisibility" Value="Collapsed">
+                <Setter TargetName="PART_Corner" Property="Visibility" Value="Collapsed"/>
+            </Trigger>
+
+            <Trigger Property="wpf:ScrollViewerAssist.CornerRectangleVisibility" Value="Hidden">
+                <Setter TargetName="PART_Corner" Property="Visibility" Value="Hidden"/>
+            </Trigger>
+
+            <MultiTrigger>
+                <MultiTrigger.Conditions>
+                    <Condition Property="wpf:ScrollViewerAssist.IsAutoHideEnabled" Value="True" />                    
+                    <Condition Property="IsMouseOver" Value="True" />
```

Comment:
I am thinking we should probably add one more condition and separate out the `PART_VerticalScrollBar` and `PART_HorizontalScrollBar` into their own MultiTriggers.
```
<Condition Property="ComputedVerticalScrollBarVisibility" Value="Visible" />
```
That should eliminate the need for the Triggers below that are checking for Disabled and Hidden. It also allows for individually controlling which scrollbars get hidden/shown.

Reply:
Thank you for the recommendations